kt-android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,229 @@
+package io.sdkgen.runtime
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Point
+import android.os.Build
+import android.provider.Settings
+import android.view.WindowManager
+import com.google.gson.*
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.net.SocketTimeoutException
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.timerTask
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+@Suppress("unused")
+class SdkGenHttpClient(
+    private val baseUrl: String,
+    private val applicationContext: Context,
```

Comment:
is it possible for `applicationContext` to be optional (nullable)?

Reply:
We need context to access the sharedPreferences (where we store the deviceId), app version from packageManager, screen info and fingerprint 🤔 
Also, I don't think this is a problem... it's usual to initiate this kind of singletons on Application's onCreate method.